python/mleap/sklearn/preprocessing/data.py

```diff
@@ -87,7 +87,7 @@ def mleap_init(self, prior_tf, output_features=None):
         output_size = 1
         output_feature_name = prior_tf.output_features
 
-    class_name = "{}".format(self.__init__.im_class).split('.')[-1].replace('>','').replace("'",'')
+    class_name = "{}".format(self.__class__.__name__).split('.')[-1].replace('>','').replace("'",'')
```

Comment:
@trams with these changes, I believe the following will suffice. the result from ```im_class``` was in the format ``` <class 'mleap.sklearn.preprocessing.data.FeatureExtractor'> ``` and that's why there were the extra processing steps afterwards.

``` 
class_name = self.__class__.__name__
```

Reply:
Yes. You are right. I'll submit a new patch